src/charm.py

```diff
@@ -54,21 +54,11 @@ def __init__(self, *args: typing.Any) -> None:
         """
         super().__init__(*args)
         self._backup = BackupObserver(self)
+        self._database = DatabaseObserver(self, relation_name=synapse.SYNAPSE_DB_RELATION_NAME)
         self._irc_bridge_database = DatabaseObserver(self, relation_name="irc-bridge-database")
-        self._database = DatabaseObserver(self)
         self._saml = SAMLObserver(self)
         self._smtp = SMTPObserver(self)
-        try:
-            self._charm_state = CharmState.from_charm(
-                charm=self,
-                datasource=self._database.get_relation_as_datasource(),
-                irc_bridge_datasource=self._irc_bridge_database.get_relation_as_datasource(),
-                saml_config=self._saml.get_relation_as_saml_conf(),
-                smtp_config=self._smtp.get_relation_as_smtp_conf(),
-            )
-        except CharmConfigInvalidError as exc:
-            self.model.unit.status = ops.BlockedStatus(exc.msg)
-            return
```

Comment:
I'm confused - if this is removed, where is `_charm_state` set instead?

Reply:
answered below